Foundation/URLSession/http/HTTPURLProtocol.swift

```diff
@@ -330,12 +330,13 @@ internal extension _HTTPURLProtocol {
         case stream(InputStream)
     }
 
-    func failWith(errorCode: Int, request: URLRequest) {
+    func failWith(errorCode: Int, errorDescription: String?, request: URLRequest) {
```

Comment:
Would it be better if this function took a `URLErrorInfo` instead?

Reply:
This doesn’t realy pan out as you would be having to create a URLErrrorInfo at the last minute in two out of three of the places where this function is called. Better to leave it as an extra argument. URLErrorInfo is mostly to covery the into through the various callbacks.